### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by wikkyk**
1. Ported the api_client tests that didn't get merged into the public release (from @65278)
2. More taskservice tests
3. More vmservice tests
4. Configured goimports + a reformatting of offending files

### CODE REVIEW

**Comment by wikkyk** [comment]
Stats so far:

-cmd 6.0%
+cmd 6.7%

-internal/controller 24.6%
+internal/controller 26.3%

-internal/service/taskservice 0.0%
+internal/service/taskservice 4.5%

-internal/service/vmservice 72.1%
+internal/service/vmservice 75.6%

-pkg/proxmox/goproxmox 35.8%
+pkg/proxmox/goproxmox 74.8%

File: internal/service/vmservice/bootstrap_test.go [hunk 1]
```diff
@@ -109,11 +111,65 @@ func TestReconcileBootstrapData_UpdateStatus(t *testing.T) {
...
 func TestGetBootstrapData_MissingSecretName(t *testing.T) {
 	machineScope, _, _ := setupReconcilerTest(t)
 
 	data, err := getBootstrapData(context.Background(), machineScope)
 	require.Error(t, err)
+	require.Equal(t, err.Error(), "machine has no bootstrap data")
+	require.Nil(t, data)
+}
+
+func TestGetBootstrapData_MissingSecretNotName(t *testing.T) {
```
Comment (mcbenjemaa) [inline internal/service/vmservice/bootstrap_test.go:144]: isn't this similar to `estGetBootstrapData_MissingSecretName`

File: pkg/proxmox/goproxmox/api_client_test.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 	"testing"
 
 	"github.com/go-logr/logr"
+	capmox "github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/proxmox"
```
Comment (mcbenjemaa) [inline pkg/proxmox/goproxmox/api_client_test.go:26]: please goimports file

File: internal/service/vmservice/bootstrap_test.go (hunk 1 above)
Comment (wikkyk) [reply internal/service/vmservice/bootstrap_test.go:144]: It is similar but it tests a different codepath: `TestGetBootstrapData_MissingSecretName` tests the `if scope.Machine.Spec.Bootstrap.DataSecretName == nil {` whereas my new test `TestGetBootstrapData_MissingSecretNotName` tests the  `if err := scope.GetBootstrapSecret(ctx, secret); err != nil {` that follows.

File: pkg/proxmox/goproxmox/api_client_test.go (hunk 2 above)
Comment (wikkyk) [reply pkg/proxmox/goproxmox/api_client_test.go]: added a lint for this

File: internal/controller/proxmoxmachine_controller.go [hunk 3]
```diff
@@ -37,6 +36,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	"github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/proxmox"
+
```
Comment (mcbenjemaa) [inline internal/controller/proxmoxmachine_controller.go:40]: unnecessary empty line

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -16,12 +16,26 @@ jobs:
       - uses: actions/setup-go@v5
         with:
           go-version-file: go.mod
-          cache: false
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v6
         with:
           version: v1.57
 
+  goimports:
```
Comment (mcbenjemaa) [inline .github/workflows/lint.yml:24]: I don't think we need this.
golangci already enabled the goimports checks

Comment (wikkyk) [reply .github/workflows/lint.yml:24]: It's not failing the lint for failed goimports.

Comment (mcbenjemaa) [reply .github/workflows/lint.yml:24]: Can you at least move it to Makefile target?

Comment (wikkyk) [reply .github/workflows/lint.yml]: I figured it out. I configured golangci to get the desired behaviour and removed the new action.

**Review by mcbenjemaa — APPROVED** [review]
LGTM

**Pull request merged**